### ISSUE TITLE
Load function requires an object for the return value

### ISSUE THREAD

**Issue by aburymichael** [sveltejs/kit#1444]
**Describe the bug**
When returning nothing in a load function (as perceived to be permitted [by the docs](https://kit.svelte.dev/docs#loading-output) ) an error is thrown that doesn't pin point the actual issue - which may not even need to be one - which is that I wrote `return;` at the end of the load function instead of `return {};`

**Logs**
Cannot read property 'error' of undefined
TypeError: Cannot read property 'error' of undefined
    at normalize (node_modules/@sveltejs/kit/dist/ssr.js:517:13)
    at load_node (node_modules/@sveltejs/kit/dist/ssr.js:829:11)
    at async respond_with_error (node_modules/@sveltejs/kit/dist/ssr.js:904:17)
    at async respond$1 (node_modules/@sveltejs/kit/dist/ssr.js:1119:13)
    at async render_page (node_modules/@sveltejs/kit/dist/ssr.js:1186:20)
    at async render (node_modules/@sveltejs/kit/dist/ssr.js:1537:10)
    at async Object.handle (/src/hooks:26:19)
    at async respond (node_modules/@sveltejs/kit/dist/ssr.js:1510:10)
    at async Immediate.<anonymous> (node_modules/@sveltejs/kit/dist/chunks/index.js:3305:23)

**To Reproduce**
in any `@type {import('@sveltejs/kit').Load}` function, change current return value to `return;`

**Expected behavior**
To set default object with status 200

**Information about your SvelteKit Installation:**

<details>
  <summary>Diagnostics</summary>
System:
    OS: macOS 11.3.1
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Memory: 172.27 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.15.4 - ~/.nvm/versions/node/v14.15.4/bin/node
    npm: 7.12.1 - ~/.nvm/versions/node/v14.15.4/bin/npm
  Browsers:
    Chrome: 90.0.4430.212
    Firefox: 87.0
    Safari: 14.1
  npmPackages:
    @sveltejs/kit: next => 1.0.0-next.107 
    svelte: ^3.34.0 => 3.38.2 
</details>

**Severity**
Not severe, easily fixable.

**Additional context**
If this is by design, perhaps the docs could be updated to reflect this? With it currently saying that all keys are optional you'd assume you wouldn't have to pass an empty object to an all optional object but rather nothing at all.

Thanks for all the hard work!

**Comment by Conduitry** [comment]
Do you have a more specific reproduction for what you're seeing? Returning a falsy value from a `load` function should result in a 404 (see https://kit.svelte.dev/docs#routing-advanced-fallthrough-routes). This error sounds like you've written an `__error.svelte` component that is trying to handle it, but its own `load` function isn't properly passing the `error` object as a prop to the component.

**Comment by aburymichael** [comment]
Seems to be directly related to using the load function within the `__layout.svelte` component, which I'm not too sure is even a thing/allowed.

I was trying to use it as a 'catch all routes' function. I used it because I was looking for a way to redirect, and the load docs actually had the ['redirect' property](https://kit.svelte.dev/docs#loading-output-redirect) listed so I thought this was the way to do it.
Looking at the [handle hook](https://kit.svelte.dev/docs#hooks-handle) it makes more sense to have the logic there - the docs are just a bit lacking here in what this is capable of doing.

I ended up moving the logic into the hooks.js as it was better suited there for authentication purposes but basically the code was this in `__layout.svelte`:

````
<script context="module">
	/**
	 * @type {import('@sveltejs/kit').Load}
	 */
	export async function load({ page, session }) {
		if (!session.person.uid && page.path !== '/login') {
			return {
				redirect: '/login',
				status: '302'
			}
		}
		return
	}
</script>
````

This threw the error I listed and if I removed the last return, it threw the same error. The only fix was to say `return {}` which seemed a bit odd to me. Bearing in mind it is a weird use case for load (hence why I've moved it) but someone might do something similar and run into the same issue.

I haven't added an `__error.svelte` component to my project and you can replicate the issue by adding the above code to `__layout.svelte` component in the demo app.

**Comment by dummdidumm** [comment]
This `__layout` thing is the key here I think. I don't know though if this considered "works as designed" and we should update the docs to make clear that a `load` function in a `__layout` file needs to return something, or if this is a bug and this should be treated as if someone did not enhance the load function at all. I'd say both: 
- It should be okay to not return anything in the load function of a `__layout` file and treat this as if there was no `load` function at all
- We should enhance documentation on https://kit.svelte.dev/docs#layouts and provide a short info that you can use the `load` function for passing context and for doing work like guarding against unallowed access.

**Comment by Conduitry** [comment]
Ah I see. I think this is probably a bug, yeah, but I'll let Rich weigh in. For route fallthrough purposes, I can't see a reason why you'd want different behavior when returning falsy from a layout load.

**Comment by benmccann** [comment]
The error message for this is dramatically better now. I think it got updated when I turned on strict typing for TypeScript in the most recent releases. Now it says:

> Error: /src/routes/__layout.svelte - load must return a value except for page fall through

The main issue remaining here is that it just shows it on a blank page instead of the normal error page:

https://github.com/sveltejs/kit/blob/2a1e9795ad9773e04669a7b9d0234caa0b688ade/packages/kit/src/runtime/server/page/load_node.js#L269

But I guess that happens because since this error is on a layout page it occurs on the error page and so since the error page can't be rendered it just shows a blank page with the error. I don't have any ideas how to do that differently, so am going to call this one solved
